plan/physical_plan_builder.go

```diff
@@ -61,6 +62,14 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 		ts.readOnly = true
 	}
 
+	infoSchema, ok := getInfoSchema(p.ctx)
```

Comment:
@tiancaiamao, 
Thank you for the valuable comments. You mean that txn.InfoSchema should not be used in the plan generation phase because it is not safe. Please confirm what I understand.